### PR TITLE
Mongo db implementation [WIP]

### PR DESCRIPTION
**Pull request by gogl92**
Needs improvements but a general idea 🚀

For #13

### CODE REVIEW

**Comment by adhocore** [comment]
thank you 👍 
composer JSON should not be necessary

File: 8.1.Dockerfile [hunk 1]
```diff
@@ -37,6 +38,22 @@ RUN \
...
   && rm -rf /var/cache/apk/* /tmp/* /var/tmp/* /usr/share/doc/* /usr/share/man/*
+  
+### MongoDB
+   RUN set -x && \
+       apk update && \
+       apk add \
+    	   bzip2 \
+    	   xz
+
+    RUN echo "http://dl-cdn.alpinelinux.org/alpine/v$ALPINE_VERSION/main" >> /etc/apk/repositories
```
Comment (adhocore) [inline 8.1.Dockerfile:49]: can it be merged to above RUN?

File: 8.1.Dockerfile [hunk 2]
```diff
@@ -37,6 +38,22 @@ RUN \
...
+    RUN echo "http://dl-cdn.alpinelinux.org/alpine/v$ALPINE_VERSION/community" >> /etc/apk/repositories
+    RUN apk update
+    RUN apk add mongodb \
+                mongodb-tools
+
+# create mongodb directory
+RUN mkdir -p /data/db
```
Comment (adhocore) [inline 8.1.Dockerfile:56]: i prefer single RUN for one job :)

File: docker-entrypoint.sh [hunk 3]
```diff
@@ -86,4 +89,29 @@ if [ "$DISABLE_PGSQL" != "YES" ] && [ ! -f /run/postgresql/.init ]; then
...
+                   { role: \"dbAdminAnyDatabase\", db: \"admin\" },
+                   { role: \"clusterAdmin\", db: \"admin\" }
+                ]
+          }
+)" > /data/admin.js;
+mongod --dbpath /data/db run &
+mongopid
+
+# Wait for the mongo server to come up
+sleep 25
```
Comment (adhocore) [inline docker-entrypoint.sh:110]: that's pretty long sleep duration :D

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -86,4 +89,29 @@ if [ "$DISABLE_PGSQL" != "YES" ] && [ ! -f /run/postgresql/.init ]; then
...
+    pwd: \"$MONGODB_PASSWORD\",
+         roles: [
+                   { role: \"userAdminAnyDatabase\", db: \"admin\" },
+                   { role: \"readWriteAnyDatabase\", db: \"admin\" },
+                   { role: \"dbAdminAnyDatabase\", db: \"admin\" },
+                   { role: \"clusterAdmin\", db: \"admin\" }
+                ]
+          }
+)" > /data/admin.js;
+mongod --dbpath /data/db run &
```
Comment (adhocore) [inline docker-entrypoint.sh:106]: should be moved to supervisor (pls check how other binaries are run in this repo)

**Comment by adhocore** [comment]
maybe need to create new PR?

**Comment by gogl92** [comment]
@adhocore yeah sorry I've been busy these last months, I was thinking maybe I need an sponsor for this PR.

**Comment by adhocore** [comment]
oh ok then 😃 and thanks :)

**Pull request closed without merging**